willow/plugins/pillow.py

```diff
@@ -214,6 +215,14 @@ def save_as_heic(self, f, quality=80, lossless=False):
             self.image.save(f, 'HEIF', quality=quality)
         return HeicImageFile(f)
 
+    @Image.operation
+    def save_as_avif(self, f, quality=80, lossless=False):
+        if lossless:
+            self.image.save(f, 'AVIF', quality=-1, chroma=444)
+        else:
+            self.image.save(f, 'AVIF', quality=quality)
```

Comment:
Might be one for later but I’m surprised we’d not make the chroma subsampling configurable 🤔 I don’t know whether the default is `4:2:2` or `4:2:0` but I could imagine scenarios where we’d want to change it.

Reply:
note to self: https://pillow-heif.readthedocs.io/en/latest/options.html mentions chroma=444 is to be using lossless encoding.

there's no stopping us from doing

```python
@Image.operation
def save_as_avif(self, f, quality=80, lossless=False, chroma: int = 420):
    if lossless:
        self.image.save(f, 'AVIF', quality=-1, chroma=444)
    else:
        self.image.save(f, 'AVIF', quality=quality, chroma=chroma)
```

at the same time, we will want to document some best practices on chroma and to be honest this is not something I fully understand (yet)

Found https://github.com/bigcat88/pillow_heif/blob/891d3fcef08eb03066f557ccef8fdc2492fd1c40/CHANGELOG.md?plain=1#L191 which mentions 420 (`4:2:0`) is the default

It would be good to document the quality kwarg, and what the different values mean (Possible values: None, -1, range(0-100). Set -1 for lossless quality or from 0 to 100, where 0 is lowest and 100 is highest.)

If we are to support pillow-avif-plugin as an alternative to pillow-heif, then this becomes more complicated - https://github.com/fdintino/pillow-avif-plugin/blob/master/src/pillow_avif/AvifImagePlugin.py#L130-L137. The latter plugin has a corresponding PR for Pillow - https://github.com/python-pillow/Pillow/pull/5201